oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/db/OSHDBJdbc.java

```diff
@@ -80,19 +80,19 @@ public OSHDBJdbc prefix(String prefix) {
 
   @Override
   public String metadata(String property) {
-    try (PreparedStatement stmt = connection.prepareStatement(
-        "SELECT value from " + TableNames.T_METADATA.toString(this.prefix()) + " where key=?"
-    )) {
+    var table = TableNames.T_METADATA.toString(this.prefix());
+    var selectSql = String.format("select value from %s where key=?", table);
+    try (PreparedStatement stmt = connection.prepareStatement(selectSql)) {
       stmt.setString(1, property);
-      ResultSet result = stmt.executeQuery();
-      if (result.next()) {
-        return result.getString(1);
-      } else {
-        return null;
+      try (var result = stmt.executeQuery()) {
+        if (result.next()) {
+          return result.getString(1);
+        }
       }
     } catch (SQLException ignored) {
-      return null;
+      // ignored
```

Comment:
What about throwing a `OSHDBTableNotFoundException` here instead of silently ignoring the error (which wold most likely happen because an OSHDB file doesn't have metadata table). Especially if we want to use this in the future to check for compatibility of a given OSHDB file, we need a valid metadata table anyway.

Reply:
👍🏽  makes sense, I update it.

Reply:
What do you think about to return an empty string instead of null in case there is no properties that match.

Reply:
hmm, not a big fan because the value stored in the table could actually be an empty string. But the method could return an empty `Optional<String>` if a value was absent.

Reply:
> What about throwing a `OSHDBTableNotFoundException` here instead of silently ignoring the error (which wold most likely happen because an OSHDB file doesn't have metadata table). Especially if we want to use this in the future to check for compatibility of a given OSHDB file, we need a valid metadata table anyway.

It could also be the case that the database/connection in this moment, is not valid any more, so I throw better a OSHDBException and pass the SQLExcetption to it.

Reply:
> hmm, not a big fan because the value stored in the table could actually be an empty string. But the method could return an empty `Optional<String>` if a value was absent.

fine for me, so I don't change this now.

Reply:
follow up ticket: #421